### PR TITLE
[BEAM-1395,BEAM-1396] Remove needless assumptions and complexity from GABWViaOutputBufferDoFn

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @amitsela / @dhalperi

**Comment by kennknowles** [comment]
https://builds.apache.org/view/Beam/job/beam_PostCommit_Java_RunnableOnService_Dataflow/2209/

File: runners/core-java/src/main/java/org/apache/beam/runners/core/GroupAlsoByWindowsViaOutputBufferDoFn.java [hunk 1]
```diff
@@ -77,25 +75,8 @@ public void processElement(ProcessContext c) throws Exception {
...
-      timerInternals.advanceSynchronizedProcessingTime(Instant.now());
-
-      // Fire all the eligible timers.
-      fireEligibleTimers(timerInternals, reduceFnRunner);
-
-      // Leave the output watermark undefined. Since there's no late data in batch mode
-      // there's really no need to track it as we do for streaming.
-    }
+    // Process the elements.
+    reduceFnRunner.processElements(c.element().getValue());
```
Comment (dhalperi) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/GroupAlsoByWindowsViaOutputBufferDoFn.java]: (ignore as you see fit!)

Personally, I would appreciate the functional and formatting changes separated. If I look carefully, I think this is the main change and the other half of the diff is just whitespace.

**Comment by dhalperi** [comment]
LGTM but @amitsela should be the one to sign off. Thanks Kenn, this is very nice. And thanks for the excellent info in JIRA description.

**Comment by dhalperi** [comment]
(Maybe also tag with BEAM-1396 ?)

**Comment by amitsela** [comment]
LGTM. 
I'm more easy going than Dan on the commits, but I agree with the tagging - should be [BEAM-1396] or we will have 2 PRs (& #1922) tagged the same.

File: runners/core-java/src/main/java/org/apache/beam/runners/core/GroupAlsoByWindowsViaOutputBufferDoFn.java (hunk 1 above)
Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/GroupAlsoByWindowsViaOutputBufferDoFn.java]: Done

**Comment by amitsela** [comment]
This is a flake. Hope to remove those 2 soon.

**Comment by kennknowles** [comment]
OK good. I had peeled off the formatting and spelling errors from the main change, and trusted local mergebot since this doesn't touch the core SDK (hence no transport issues).

**Pull request merged**